draco/analysis/mapmaker.py

```diff
@@ -70,58 +73,79 @@ def find_key(key_list, key):
             except ValueError:
                 return None
 
-        # Figure out mapping between the frequencies
-        bt_freq = self.beamtransfer.telescope.frequencies
-        mm_freq = mmodes.index_map['freq']['centre']
+        if self.basis == 'mmodes':
```

Comment:
Don't almost all the things in both clauses of this `if ... else ...` block need doing for both `mmodes` and `svdmodes`? Or at the very least they can be safely done for both.

Reply:
In other words remove this if-else structure if you can.

Reply:
the if else structure can be mostly removed thats correct. 
line 79:  `mm_freq = modes.index_map['freq']['centre']`
line 80: `freq_ind = [find_key(bt_freq, mf) for mf in mm_freq]`
these two lines though are not working for svd modes because their index map has only m, and mode. we could add an frequency axis, although people have to aware that the number of svdmodes is different for every frequency and flatted on the `mode` axis. opinions?